File: Back-End/credits/src/main/java/com/system/credits/dto/NewUser.java
package com.system.credits.dto;

import javax.validation.constraints.NotBlank;
import java.sql.Date;
import java.util.HashSet;
import java.util.Set;

public class NewUser {
    @NotBlank
    private String run;
    @NotBlank
    private String document;
    @NotBlank
    private String email;
    @NotBlank
    private String password;
    @NotBlank
    private String firstName;
    @NotBlank
    private String middleName;
    @NotBlank
    private String paternalLastName;
    @NotBlank
    private String maternalLastName;

    private String dateOfBirth;

    private Integer phone;
    @NotBlank
    private String region;
    @NotBlank
    private String city;
    @NotBlank
    private String address;
    private Set<String> roles = new HashSet<>();

    public String getRun() {
        return run;
    }

    public void setRun(String run) {
        this.run = run;
    }

    public String getDocument() {
        return document;
    }

    public void setDocument(String document) {
        this.document = document;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getMiddleName() {
        return middleName;
    }

    public void setMiddleName(String middleName) {
        this.middleName = middleName;
    }

    public String getPaternalLastName() {
        return paternalLastName;
    }

    public void setPaternalLastName(String paternalLastName) {
        this.paternalLastName = paternalLastName;
    }

    public String getMaternalLastName() {
        return maternalLastName;
    }

    public void setMaternalLastName(String maternalLastName) {
        this.maternalLastName = maternalLastName;
    }

    public String getDateOfBirth() {
        return dateOfBirth;
    }

    public void setDateOfBirth(String dateOfBirth) {
        this.dateOfBirth = dateOfBirth;
    }

    public Integer getPhone() {
        return phone;
    }

    public void setPhone(Integer phone) {
        this.phone = phone;
    }

    public String getRegion() {
        return region;
    }

    public void setRegion(String region) {
        this.region = region;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Set<String> getRoles() {
        return roles;
    }

    public void setRoles(Set<String> roles) {
        this.roles = roles;
    }
}

File: Front-End/src/app/entity/new-user.ts
export class NewUser {
    userData: string
    run: string;
    document: string;
    email: string;
    password: string;
    firstName: string;
    middleName: string;
    paternalLastName: string;
    maternalLastName: string;
    dateOfBirth: string;
    phone: string;
    region: string;
    city: string;
    address: string;
    //authorities: string[];
    
    constructor(
        run: string,
        document: string,
        email: string,
        password: string,
        firstName: string,
        middleName: string,
        paternalLastName: string,
        maternalLastName: string,
        dateOfBirth: string,
        phone: string,
        region: string,
        city: string,
        address: string
    ) {
        this.run = run
        this.document = document
        this.email = email
        this.password = password
        this.firstName = firstName
        this.middleName = middleName
        this.paternalLastName = paternalLastName
        this.maternalLastName = maternalLastName
        this.dateOfBirth = dateOfBirth
        this.phone = phone
        this.region = region
        this.city = city
        this.address = address
    }
}
